cmd/podman/pod_ps.go

```diff
@@ -0,0 +1,562 @@
+		}, nil
+	case "status":
+		if !util.StringInSlice(filterValue, []string{"stopped", "running", "paused", "exited", "dead", "created"}) {
+			return nil, errors.Errorf("%s is not a valid pod status", filterValue)
+		}
+		return func(p *libpod.Pod) bool {
+			ctrs, err := p.AllContainers()
+			if err != nil {
+				return false
+			}
+			status, err := getPodStatus(ctrs)
+			if err != nil {
+				return false
+			}
+			if strings.ToLower(status) == filterValue {
+				return true
+			}
+			return false
+		}, nil
+	}
+	return nil, errors.Errorf("%s is an invalid filter", filter)
+}
+
+// generate the template based on conditions given
+func genPodPsFormat(c *cli.Context) string {
+	format := ""
+	if c.String("format") != "" {
+		// "\t" from the command line is not being recognized as a tab
+		// replacing the string "\t" to a tab character if the user passes in "\t"
+		format = strings.Replace(c.String("format"), `\t`, "\t", -1)
+	} else if c.Bool("quiet") {
+		format = formats.IDString
+	} else {
+		format = "table {{.ID}}\t{{.Name}}\t{{.Status}}"
+		if c.Bool("cgroup") {
+			format += "\t{{.Cgroup}}\t{{.UsePodCgroup}}"
+		}
+		if c.Bool("ctr-names") || c.Bool("ctr-ids") || c.Bool("ctr-status") {
+			format += "\t{{.ContainerInfo}}"
+		} else {
+			format += "\t{{.NumberOfContainers}}"
+		}
+	}
+	return format
+}
+
+func podPsToGeneric(templParams []podPsTemplateParams, JSONParams []podPsJSONParams) (genericParams []interface{}) {
+	if len(templParams) > 0 {
+		for _, v := range templParams {
+			genericParams = append(genericParams, interface{}(v))
+		}
+		return
+	}
+	for _, v := range JSONParams {
+		genericParams = append(genericParams, interface{}(v))
+	}
+	return
+}
+
+// generate the accurate header based on template given
+func (p *podPsTemplateParams) podHeaderMap() map[string]string {
+	v := reflect.Indirect(reflect.ValueOf(p))
+	values := make(map[string]string)
+
+	for i := 0; i < v.NumField(); i++ {
+		key := v.Type().Field(i).Name
+		value := key
+		if value == "ID" {
+			value = "Pod" + value
+		}
+		values[key] = strings.ToUpper(splitCamelCase(value))
+	}
+	return values
+}
+
+func sortPodPsOutput(sortBy string, psOutput podPsSorted) (podPsSorted, error) {
+	switch sortBy {
+	case "id":
+		sort.Sort(podPsSortedId{psOutput})
+	case "name":
+		sort.Sort(podPsSortedName{psOutput})
+	case "number":
+		sort.Sort(podPsSortedNumber{psOutput})
+	case "status":
+		sort.Sort(podPsSortedStatus{psOutput})
+	default:
+		return nil, errors.Errorf("invalid option for --sort, options are: id, names, or number")
+	}
+	return psOutput, nil
+}
+
+// getPodTemplateOutput returns the modified container information
+func getPodTemplateOutput(psParams []podPsJSONParams, opts podPsOptions) ([]podPsTemplateParams, error) {
+	var (
+		psOutput []podPsTemplateParams
+	)
+
+	for _, psParam := range psParams {
+		podID := psParam.ID
+		var ctrStr string
+
+		truncated := ""
+		if !opts.NoTrunc {
+			podID = shortID(podID)
+			if len(psParam.CtrsInfo) > 10 {
```

Comment:
It might be better to make these numbers a const. @mheon WDYT?

Reply:
Good catch, SGTM